lib/models/repository-states/present.js

```diff
@@ -388,13 +389,27 @@ export default class Present extends State {
 
   getStatusBundle() {
     return this.cache.getOrSet(Keys.statusBundle, async () => {
-      const bundle = await this.git().getStatusBundle();
-      const results = await this.formatChangedFiles(bundle);
-      results.branch = bundle.branch;
-      if (!results.branch.aheadBehind) {
-        results.branch.aheadBehind = {ahead: null, behind: null};
+      try {
+        const bundle = await this.git().getStatusBundle();
+        const results = await this.formatChangedFiles(bundle);
+        results.branch = bundle.branch;
+        if (!results.branch.aheadBehind) {
+          results.branch.aheadBehind = {ahead: null, behind: null};
+        }
+        return results;
+      } catch (err) {
+        if (err instanceof LargeRepoError) {
+          this.transitionTo('TooLarge');
+          return {
+            branch: {},
+            stagedFiles: [],
+            unstagedFiles: [],
+            mergeConflictFiles: [],
+          };
```

Comment:
Ah, so we bail out entirely if a repo is too large? I was wondering if we'd be able to do some kind of partial status in that case with the streaming API. That's probably too error-prone though.